.clang-format

```diff
@@ -1,56 +1,88 @@
-AccessModifierOffset: -2
-AlignEscapedNewlinesLeft: false
-AlignTrailingComments: false
-AllowAllParametersOfDeclarationOnNextLine: false
-AllowShortFunctionsOnASingleLine: true
-AllowShortIfStatementsOnASingleLine: false
-AllowShortLoopsOnASingleLine: false
-AlwaysBreakBeforeMultilineStrings: false
-AlwaysBreakTemplateDeclarations: true
-BinPackParameters: false
-BreakBeforeBinaryOperators: false
-BreakBeforeBraces: Allman
-BreakBeforeTernaryOperators: false
-BreakConstructorInitializersBeforeComma: false
-ColumnLimit: 115
-CommentPragmas: ''
-KeepEmptyLinesAtTheStartOfBlocks: false
-ConstructorInitializerAllOnOneLineOrOnePerLine: false
-ConstructorInitializerIndentWidth: 4
-ContinuationIndentWidth: 2
-Cpp11BracedListStyle: true
-DerivePointerBinding: false
-IndentCaseLabels: false
-IndentFunctionDeclarationAfterType: false
-IndentWidth: 4
+﻿---
+AccessModifierOffset: '-2'
+AlignAfterOpenBracket: AlwaysBreak
+AlignConsecutiveMacros: 'true'
+AlignConsecutiveAssignments: 'true'
+AlignConsecutiveDeclarations: 'false'
+AlignEscapedNewlines: Right
+AlignOperands: 'false'
+AlignTrailingComments: 'false'
+AllowAllArgumentsOnNextLine: 'false'
+AllowAllConstructorInitializersOnNextLine: 'false'
+AllowAllParametersOfDeclarationOnNextLine: 'false'
+AllowShortBlocksOnASingleLine: 'false'
+AllowShortCaseLabelsOnASingleLine: 'false'
+AllowShortFunctionsOnASingleLine: Empty
+AllowShortIfStatementsOnASingleLine: Never
+AllowShortLambdasOnASingleLine: Empty
+AllowShortLoopsOnASingleLine: 'false'
+AlwaysBreakAfterDefinitionReturnType: None
+AlwaysBreakAfterReturnType: None
+AlwaysBreakBeforeMultilineStrings: 'false'
+AlwaysBreakTemplateDeclarations: 'Yes'
+BinPackArguments: 'false'
+BinPackParameters: 'false'
+BreakBeforeBraces: Attach
```

Comment:
I think it's easier to read `Attach`:

```cpp
try {
  foo();
} catch () {
}

void foo();

void foo() {
  int aaaa = 12;
  int b = 23;
  int ccc = 23;
}

class foo {};

if (foo()) {
} else {
}

enum X : int { A, B };
```

than `Allman`:

```cpp
try
{
  foo();
}
catch ()
{
}

void foo();

void foo()
{
  int aaaa = 12;
  int b = 23;
  int ccc = 23;
}

class foo
{
};

if (foo())
{
}
else
{
}

enum X : int
{
  A,
  B
};
```

since in `Attach`, the blocks created by `{}` are "attached" to the context as opposed to in `Allman` where it is separated from the context by a newline break. I think `Attach` helps in readability.

Reply:
I actually prefer Allman as it gives more whitespace in the file which I think helps with readability,  In some situations I can see it working better like the above example but in general I prefer the extra whitespace,  I have seen code that uses it and it can get really cluttered unless you manually add whitespace lines    I wouldn't want to make this change now though regardless of what we ultimately decide as this one would cause huge changes in the code base, so if we were to make it I don't think we should do that until the 3.0 release which would be a disconnect anyway.